appserver/monitoring-console/core/src/main/java/fish/payara/monitoring/store/InMemoryMonitoringDataRepository.java

```diff
@@ -101,6 +105,8 @@
      * The topic name used to share data of instances with the DAS.
      */
     private static final String MONITORING_DATA_TOPIC_NAME = "payara-monitoring-data";
+    private static final Logger LOGGER = Logger.getLogger("monitoring-console-core");
+    private final Set<String> sourcesFailingBefore = new HashSet<>();
```

Comment:
The reasoning for using a `HashSet` and not a concurrent one was that the collection is run by a single worker thread. As we use a pool I later thought we have no guarantees that this is the same thread. Might be better to use concurrent nonetheless.